### ISSUE TITLE
UDA Control Unit Test has Unexpected Iteration Pattern

### ISSUE THREAD

**Issue by bska** [OPM/opm-common#2483]
The [UDQ_USAGE](https://github.com/OPM/opm-common/blob/76fdcc298551bfb7b692f20902df3c6e94fba090/tests/parser/UDQTests.cpp#L1481) unit test has the following loop
```C++
for (std::size_t index = 0; index < usage.IUAD_size(); index++) {
    const auto& record = usage[index];
    BOOST_CHECK_EQUAL(record.input_index, 0U);

    if (index == 0)
        BOOST_CHECK(record.control == UDAControl::WCONPROD_ORAT);
    else
        BOOST_CHECK(record.control == UDAControl::WCONPROD_GRAT);

    index += 1;
}
```
This increments `index` twice, once in the loop control and once in the body.  Is that really the expected pattern?  If it is, might it not make sense to increment the `index` by `2` in the loop control instead?

**Comment by joakim-hove** [comment]
Looks like a bug.

**Comment by bska** [comment]
> Looks like a bug.

Cool.  I've made a fix in #2484.
